### PR TITLE
os/board/rtl8721csm: Remove unneeded RTL mpu setting

### PR DESCRIPTION
**Pull request by ziyik**
- Removed 3 mpu setting "close 216K irom_ns cache", "close 80K drom_ns cache" and "set 1KB retention ram no-cache"
- Added PSRAM mpu setting "set PSRAM Memory Write-Back", to prevent data confusion issue in PSRAM

### CODE REVIEW

File: os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c [hunk 1]
```diff
@@ -938,6 +938,10 @@ u32 app_mpu_nocache_init(void)
 	mpu_region_config mpu_cfg;
 	u32 mpu_entry = 0;
 
+	/*	Below MPU setting is No cache init
```
Comment (amandeepc) [inline os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:941]: Can you please use below comment format for better readability:
/*
 *
 */

Comment (amandeepc) [reply os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:941]: Also, can you please rephrase it to make it more understandable.

File: os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c [hunk 2]
```diff
@@ -953,7 +957,12 @@ u32 app_mpu_nocache_init(void)
 	if (mpu_cfg.region_size >= 32) {
 		mpu_region_cfg(mpu_entry, &mpu_cfg);
 	}
-
+#ifndef CONFIG_PLATFORM_TIZENRT_OS
+/*	For TizenRT  below 3 MPU seting is not use
```
Comment (amandeepc) [inline os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:961]: same as above

Comment (amandeepc) [reply os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:961]: ** setting

File: os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c [hunk 3]
```diff
@@ -953,7 +957,12 @@ u32 app_mpu_nocache_init(void)
...
+	"close 216K irom_ns cache" and "close 80K drom_ns cache" is to prevent ROM code
+	to access data through Cache, by testing to remove it perforamnce have not drop.
```
Comment (thapav) [inline os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:963]: performance*

Comment (thapav) [reply os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:963]: Also, do you mean that :
by removing these 3 settings, performance does not drop ?

Comment (ziyik) [reply os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c:963]: Yes, there is not much performance drop after removing these 3 MPU setting.
I will rephrase it later.

File: os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c (hunk 1 above)
Comment (ziyik) [reply os/board/rtl8721csm/src/component/soc/realtek/amebad/fwlib/ram_hp/rtl8721dhp_app_start.c]: I have rephrase it and changes the comment format.
Please let me know is there is any problem.

**Review by thapav — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]
